### DISCUSSION TITLE
How to minimize bundle size with client preset on types file?

### DISCUSSION THREAD

**Discussion by samoralespu** [dotansimha/graphql-code-generator#10016] [Q&A]
Hi, I created a codegen file as follows:

```
const config: CodegenConfig = {
  schema: "http://localhost:5000",
  // this assumes that all your source files are in a top-level `src/` directory - you might need to adjust this to your file structure
  documents: ["src/**/*.{ts,tsx}"],
  generates: {
    "./src/types/generated/": {
      preset: "client",
      plugins: [],
      presetConfig: {
        useTypeImports: true,
        gqlTagName: "gql",
      },
    },
  },
  ignoreNoDocuments: false,
};
```

It generates all files correctly, the issue waS when I tried to build my project, the bundle size was too big for the files generated:

![image](https://github.com/dotansimha/graphql-code-generator/assets/30605010/ec2544fa-96ec-4081-932f-4f2fe165ba6f)

So I decided to minimize the bundle size with babel as the guide sugested:

```
module.exports = {
  plugins,
  // presets: ['react-app'],
  // plugins: [[babelOptimizerPlugin, { artifactDirectory: './src/types/generated', gqlTagName: 'gql' }]],
}
```
The issue is that it only reducedd the gql.ts file, but the graphql.ts (the types file) is still there:

![image](https://github.com/dotansimha/graphql-code-generator/assets/30605010/80c7fd73-146a-461e-8625-7eae20f85888)

How can I minimize this bundle size?

**Comment by sep2** [comment]
This works for me https://github.com/dotansimha/graphql-code-generator/issues/9988#issuecomment-2481309224
